### PR TITLE
feat: MS planner responder integration

### PR DESCRIPTION
**Pull request by yashmehrotra**
Opened a draft PR for early suggestions/feedback. Few parts are yet to be completed.

### CODE REVIEW

File: responder/msplanner/msplanner.go [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+		convBody := models.NewConversationThread()
+		topic := fmt.Sprintf("Conversation thread topic for taskID: %s", taskID)
+		convBody.SetTopic(&topic)
+
+		convBody.SetPosts([]models.Postable{post})
+
+		result, err := c.client.GroupsById(c.groupID).Threads().Post(convBody)
+		if err != nil {
+			fmt.Println("error creating Conversation thread")
+			return openDataError(err)
```
Comment (moshloop) [inline responder/msplanner/msplanner.go:106]: Don't log and return an error

File: responder/msplanner/msplanner.go [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+		}
+
+		// TODO: For debugging
+		fmt.Println("Conversation thread created")
+		fmt.Println(result)
```
Comment (moshloop) [inline responder/msplanner/msplanner.go:111]: use commons/logger trace

File: responder/msplanner/msplanner.go (hunk 1 above)
Comment (yashmehrotra) [reply responder/msplanner/msplanner.go:106]: Yeah, this is just for debugging on local, will remove all print statements

File: responder/msplanner/msplanner.go (hunk 2 above)
Comment (yashmehrotra) [reply responder/msplanner/msplanner.go:111]: Yeah, this is just for debugging on local, will remove all print statements

File: events/events.go [hunk 3]
```diff
@@ -127,9 +127,12 @@ func reconcileResponderEvent(tx *gorm.DB, event api.Event) error {
 	var externalID string
 	if responder.Properties["responderType"] == "Jira" {
 		externalID, err = responderPkg.NotifyJiraResponder(responder)
-		if err != nil {
-			return err
-		}
+	} else if responder.Properties["responderType"] == "MSPlanner" {
+		externalID, err = responderPkg.NotifyMSPlannerResponder(responder)
+	}
+
```
Comment (yashmehrotra) [inline events/events.go:133]: @moshloop Should I use a `getResponderFunc(responderType string) func(responder) (string, error)` or a `switch` here or is if/else fine for now

File: events/events.go [hunk 4]
```diff
@@ -127,9 +127,12 @@ func reconcileResponderEvent(tx *gorm.DB, event api.Event) error {
 	var externalID string
 	if responder.Properties["responderType"] == "Jira" {
 		externalID, err = responderPkg.NotifyJiraResponder(responder)
-		if err != nil {
-			return err
-		}
+	} else if responder.Properties["responderType"] == "MSPlanner" {
```
Comment (Kaitou786) [inline events/events.go:130]: Can we make these `MSPlanner` and JIRA as constants and probably create a type Responders which is a string and restrict the types from there?

File: responder/events.go [hunk 5]
```diff
@@ -48,3 +50,43 @@ func NotifyJiraResponder(responder api.Responder) (string, error) {
...
+	if err = json.Unmarshal(teamSpecJson, &teamSpec); err != nil {
+		return "", err
+	}
+
+	client, err := msplanner.NewClient(
+		teamSpec.ResponderClients.MSPlanner.TenantID,
+		teamSpec.ResponderClients.MSPlanner.ClientID,
+		teamSpec.ResponderClients.MSPlanner.GroupID,
+		teamSpec.ResponderClients.MSPlanner.Username.Value,
+		teamSpec.ResponderClients.MSPlanner.Password.Value,
```
Comment (Kaitou786) [inline responder/events.go:74]: Will we be getting the value directly or do we need to first fetch it from a secret?
Like we have in canary-checker at `checks/common.go:L65`

File: responder/msplanner/msplanner.go [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+			Name:       *plan.GetTitle(),
+			Buckets:    buckets,
+			Priorities: priorities,
+		}
+	}
+
+	return config, nil
+}
+
+func openDataError(err error) error {
```
Comment (Kaitou786) [inline responder/msplanner/msplanner.go:196]: just curious why do we need to cater for it specially

File: events/events.go (hunk 3 above)
Comment (Kaitou786) [reply events/events.go:133]: I would incline towards using a switch

File: responder/msplanner/msplanner.go (hunk 6 above)
Comment (yashmehrotra) [reply responder/msplanner/msplanner.go:196]: The microsoft graph API returns specific `openData` errors where you have two extract multiple things from the error. If I just log the error as is, we'll get a `error bad status code received`

File: events/events.go (hunk 4 above)
Comment (yashmehrotra) [reply events/events.go:130]: 👍🏽

File: responder/events.go (hunk 5 above)
Comment (yashmehrotra) [reply responder/events.go:74]: We'll be fetching the secret and storing it in the database, and the database would be the source of truth. So this flow assumes that the k8s object -> database part is already done.

**Pull request merged**